sources/vault.move

```diff
@@ -0,0 +1,122 @@
+/// User can deposit/withdraw multiple fungible assets to/from different markets
+module playground::vault {
+    use std::signer;
+    use std::string;
+
+    use aptos_std::table::{Self, Table};
+    use aptos_framework::fungible_asset::{Self, FungibleAsset, Metadata};
+    use aptos_framework::object::{Self, Object};
+
+    #[test_only]
+    use aptos_framework::primary_fungible_store;
+    #[test_only]
+    use std::option;
+    use std::vector;
+
+    const ERR_UNAUTHORIZED: u64 = 1;
+    const ERR_INSUFFICIENT_FUNDS: u64 = 2;
+
+    /// Market holds user deposited assets
+    struct Market has key {
```

Comment:
You can use the resource group annotation to put this into the same resource group as object core. This would help in the future if you need to add more resources to market objects that are accessed together. These resources would be stored together in the same storage slot => higher storage and gas efficiency.

#[resource_group_member(group = aptos_framework::object::ObjectGroup)]